File: framework/languages/java/samples/java/TicTacToe.SessionGateway/Server/Play/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/server/play/gamespots/handlers/TicTacToeGameSpotActorJoinedHandler.java
package systems.zlink.samples.tictactoe.sessiongateway.server.play.gamespots.handlers;

import systems.zlink.samples.tictactoe.sessiongateway.server.play.gamespots.TicTacToeMatchRoom;

public final class TicTacToeGameSpotActorJoinedHandler {
    public TicTacToeMatchRoom handle(TicTacToeMatchRoom room, String actorId) {
        return new TicTacToeMatchRoom(room.matchId(), actorId);
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/server/play/gamespots/handlers/TicTacToeGameSpotActorJoinedHandler.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.play.gamespots.handlers

import systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.play.gamespots.TicTacToeMatchRoom

class TicTacToeGameSpotActorJoinedHandler {
    fun handle(room: TicTacToeMatchRoom, actorId: String): TicTacToeMatchRoom =
        room.copy(lastActorId = actorId)
}
